packages/web/src/pages/sign-up-page/pages/SignUpPage.tsx

```diff
@@ -91,109 +115,170 @@ export const SignUpPage = () => {
+                        aria-label='Sign up with TikTok'
+                      />
+                    </Flex>
                   </Flex>
-                </Flex>
-                <Flex
-                  direction='column'
-                  gap='l'
-                  alignItems='flex-start'
-                  w='100%'
-                >
-                  <Button
-                    variant={ButtonType.PRIMARY}
-                    type='submit'
-                    fullWidth
-                    iconRight={IconArrowRight}
-                    isLoading={isSubmitting}
+                  <Flex
+                    direction='column'
+                    gap='l'
+                    alignItems='flex-start'
+                    w='100%'
                   >
-                    {messages.signUp}
-                  </Button>
-
-                  <Text size='l'>
-                    {messages.haveAccount}{' '}
-                    {/* TODO [C-3278]: Update with Harmony Link */}
-                    <Link
-                      to={SIGN_IN_PAGE}
-                      variant='body'
-                      size='medium'
-                      strength='strong'
-                      color='secondary'
+                    <Button
+                      variant={ButtonType.PRIMARY}
+                      type='submit'
+                      fullWidth
+                      iconRight={IconArrowRight}
+                      isLoading={isSubmitting}
                     >
-                      {messages.signIn}
-                    </Link>
-                  </Text>
-                </Flex>
-              </LeftContentContainer>
-            </Form>
-          )}
-        </Formik>
-      </PageWithAudiusValues>
-    </Flex>
+                      {messages.signUp}
+                    </Button>
+
+                    <Text size='l'>
+                      {messages.haveAccount}{' '}
+                      {/* TODO [C-3278]: Update with Harmony Link */}
+                      <Link
+                        to={SIGN_IN_PAGE}
+                        variant='body'
+                        size='medium'
+                        strength='strong'
+                        color='secondary'
+                      >
+                        {messages.signIn}
+                      </Link>
+                    </Text>
+                    {showMetaMaskOption ? (
+                      <MetaMaskOption
+                        fullWidth
+                        onClick={() =>
+                          handleClickMetaMask({
+                            values,
+                            validateForm,
+                            setErrors
+                          })
+                        }
+                      />
+                    ) : null}
+                  </Flex>
+                </LeftContentContainer>
+              </Form>
+            )}
+          </Formik>
+        </PageWithAudiusValues>
+      </Flex>
+      {showMetaMaskOption ? (
```

Comment:
can we move this into the "metamaskoption"?

Reply:
yeah probs, i'd make a new component though because the other one is still used by old sign up and don't want to mess with that

Reply:
donezo